packages/@uppy/dashboard/src/Dashboard.jsx

```diff
@@ -749,6 +749,7 @@ export default class Dashboard extends UIPlugin {
     this.startListeningToResize()
     document.addEventListener('paste', this.handlePasteOnBody)
 
+    this.uppy.on('plugin-added', this.#autoDiscoverPlugins)
```

Comment:
It seems a bit much to iterate through the entire plugin list every time this event is triggered.

Reply:
Good point, refactored, thanks.